docs/diagram.md

```diff
@@ -21,16 +21,16 @@ The execution is broken down in more detail below:
 
 The agent operation breaks down into three parts:
 
-* Setup: calls the `setup()` method of all registered resources
-* Operation:
+* **Setup**: calls the `setup()` method of all registered resources
+* **Operation**:
     * Main loop (Thread 1 - Synchronous):
```

Comment:
Note, mainloop now is asynchronous, so we should explain that. And it is no longer called mainloop, just agent loop `AgentLoop`.

Reply:
Does this mean the following is no longer correct?

```
To prevent a developer from blocking the main loop with custom skill code, an execution time limit is  applied to every `Behaviour.act` and `Handler.handle` call.

By default, the execution limit is set to `0` seconds, which disables the feature. You can set the limit to a strictly positive value (e.g. `0.1` seconds) to test your AEA for production readiness. If the `act` or `handle` time exceed this limit, the call will be terminated.

An appropriate message is added to the logs in the case of some code execution being terminated.
```

Reply:
Also, just checking, are task loop and dm loop synchronous?